### DISCUSSION TITLE
Font change?

### DISCUSSION THREAD

**Discussion by Kadeluxe** [retypeapp/retype#179] [Q&A]
I would like to change the default font because current one doesn't support needed language.
Is there a simple way to do it or do I need to 'patch' `retype.css`?

**Comment by AndreyChechel** [comment]
Hi @Kadeluxe 

There is no official configuration option for changing fonts yet.

However, you can experiment with replacement of the currently used fonts in the `.retype\resources\fonts` folder.
See the 6 files prefixed with `Poppins-`. New fonts should be named exactly the same in order to work. It's more like a hack though.

Anyways, I was able to replace the fonts by:
1. placing new fonts under `{input}/resources/fonts` 
2. and setting [`server.watch.mode`](https://retype.com/configuration/project/#watchmode) config to `disk`.

We'll discuss possibility of supporting an official configuration option for that.

**Reply by Kadeluxe** [reply]
Alright, thanks for the assistance.

**Comment by geoffreymcgill** [comment]
Hi @Kadeluxe. Do you have a text sample we could test with, containing the characters that are not rendering correctly?

**Reply by Kadeluxe** [reply]
Default font doesn't really support Unicode symbols so text in some languages looks weird compared to the Latin letters.
Take a look at the screenshot: 
![image](https://user-images.githubusercontent.com/63540878/138354840-176938e2-f17b-42fd-8216-e431c6e246a1.png)

Here it uses default Poppins for Latin and Segoe UI (I'm on Windows) for the rest. It doesn't look good when you mix both Latin and non-Latin, that's why it could be useful to have an option to change the page font accordingly.

**Comment by RickCogley** [comment]
Chiming in with my two yen here, Japanese does work, but, it's just formatting as the fallback "sans serif" if you use it, because those glyphs don't exist in Poppins:

<img width="1390" alt="image" src="https://user-images.githubusercontent.com/512328/141668424-ceff18cb-1249-4969-9a70-09ea0fcbb5b3.png">

So it's dependent on the browser and OS. Some OSes will render it OK, but others will render it in a clunky / ugly way. 
 
There are alternatives such as _M PLUS 1p_ (newly released) on Google Fonts that are well designed and have the glyphs we need for Japanese: 

https://fonts.google.com/specimen/M+PLUS+1p?query=m+plus&preview.text=%E6%97%A5%E6%9C%AC%E8%AA%9E%20%E6%9D%B1%E4%BA%AC%20%E3%82%A4%E3%82%BD%E3%83%AA%E3%82%A2%20%E3%83%AA%E3%83%BC%E3%82%BF%E3%82%A4%E3%83%97%20Almost%20before%20we%20knew%20it,%20we%20had%20left%20the%20ground.&preview.text_type=custom

Furthermore, using a well-designed Japanese font will also handle the line height better and generally look more balanced. Japanese and probably Chinese or Korean, are "blocky" compared to English, so it's usually going to look better to use specific fonts for them.

**Comment by geoffreymcgill** [comment]
As of the upcoming Retype v3.0 release, we have removed the font `Poppins` and now default to system fonts.

This change will work well with the new `locale` language targeting functionality, see https://github.com/retypeapp/retype/issues/404#issuecomment-1469152692

Hope this helps.

**Comment by geoffreymcgill** [comment]
Retype v3.0 has been released and includes the new font with support for Cyrillic languages.
